joomla/joomla-cms
# Remove multilanguage related db duplicate queries

#### Description

This PR, to remove duplicate queries from db, it saves into variables the result of the db queries of the two methods moved, only running the db query if needed.

Also moves some multilanguage methods from com_languages helper `MultilangstatusHelper` to JLanguage Multilang helper `JLanguageMultilang`.
###### Performance

This adds a little performance benefit. Around 2ms to 3ms of total page generation time (in a PHP 5.6 installation).
###### Before

![image](https://cloud.githubusercontent.com/assets/9630530/12373314/4bdaa898-bc6d-11e5-9aa3-1e4e90ee04f4.png)
###### After

![image](https://cloud.githubusercontent.com/assets/9630530/12374150/de2dca8c-bc8a-11e5-80ba-c455402f8b7e.png)
#### How to test
1. Apply patch in a default Joomla install.
2. Check if a multilanguage site is working properly (frontend mod_languages works, frontend language filter works and admin multilanguage status works)
3. Check if language associations are working properly
4. Enable the debug console and see there are less queries (check before/after PR) and less `ms` taken to do all database queries.
#### Observations

Since this PR adds and deprecates some class methods, please say if anything else is needed to do.
Suggestions, code reviews and improvements are welcome.
